docs/proposals/scale_group_upgrade.md

```diff
@@ -0,0 +1,140 @@
+# Proposal Title
+
+## Description
+With the direction of moving to golden images (AMIs), there is a need to design the
+upgrade path for scale groups.
+
+## Rationale
+We currently provide an upgrade path for in-place upgrades.  We also need to provide a path
+for golden images when placed in scale groups.
+
+## Design
+The core principle here is to take advantage of the immutable infrastructure that is provided
+when working with golden images.
+
+The current playbooks in openshift ansible create autoscaling groups for infra/compute and 
+allow for custom groups as well.  This code lives in `role/openshift_aws`.
+
+Assume we are upgrading the compute node group:
+compute 123
+
+The upgrade process would look like this:
+* Create a new scale group `compute 456`
+* Scale up the new scale group to match the first groups capacity
+* Allow nodes to join the cluster by approving node CSRs
+* Set the nodes from `compute 123` as unschedulable
+* Drain the `compute 123` pods from the nodes
+* Determine if we want to role back to `compute 123` or continue on with `compute 456`.
+* Remove scale group `compute 123` or `compute 456`
+
+## Benefits
+
+There are many benefits to this approach.
+
+* Treat the scale groups as immutable infrastructure
+* Roll back strategy is as simple as scaling the old scale group back to capacity
+* Managing hosts can be done by scale group or by scale group tags
+
+## Downside
+
+* Managing the compute name or version
+* Risk of removing scale groups
+* Process surrounding acceptance of nodes (not specific to this strategy)
+
+## User Story
+As an admin of OpenShift-Ansible,
+I want to upgrade my scale groups
+so that I can receive product updates, CVE remediations, and provide a path forward for customers.
+
+## Acceptance Criteria
+* Verify that a new scale group is created
+* Verify that the old scale group properly drained
+* Verify that the old scale group is removed
+* Verify that the old launch config is removed
+* Verify there is documentation about the process
+
+## References
+
+
+## Workflow
+
+Outline of a typical workflow for an upgrade.
+* An AMI is built with a specific version of openshift
+  This is done through openshift-ansible/playbooks/aws/openshift-cluster/build_ami.yml playbook.
+  Specify the correct repository location and version via the following vars:
+  openshift_pkg_version=
+  openshift_additional_repositories=[]
```

Comment:
minor nit pick here. I'd change the order of these variables, to match what you have.

Also, this doesn't render well

Reply:
@mwoodson, Could you clarify what you mean by order?  The `openshift_additional_repositories` is the variable that manages the yum repositories inside of openshift-ansible.  If you want the latest then you don't need to specify the `openshift_pkg_version`.

Reply:
I will switch the order.